lib/cosmos/tools/packet_viewer/packet_viewer.rb

```diff
@@ -330,8 +340,21 @@ def update_tlm_items(featured_item_name = nil)
       # Update Telemetry Items
       tlm_items = []
       begin
-        System.telemetry.items(target_name, packet_name).each do |item|
-          tlm_items << [item.name, item.states, item.description]
+        @derived_row = 0
+        if @derived_last_action.isChecked
+          System.telemetry.items(target_name, packet_name).each do |item|
+            next if item.data_type == :DERIVED # Skip DERIVED so they are last
+            tlm_items << [item.name, item.states, item.description]
+            @derived_row += 1
+          end
+          System.telemetry.items(target_name, packet_name).each do |item|
+            next if item.data_type != :DERIVED # Skip anything not DERIVED
+            tlm_items << [item.name, item.states, item.description]
+          end
```

Comment:
Looping through everything twice doesn't seem ideal.  May throw all the derived items into a seperate array and then concatenate them at the end.